pkg/api/v1/types.go

```diff
@@ -582,6 +582,8 @@ type RBDVolumeSource struct {
 	// Defaults to false.
 	// More info: http://releases.k8s.io/HEAD/examples/rbd/README.md#how-to-use-it
 	ReadOnly bool `json:"readOnly,omitempty"`
+	// Optional: Sidecar is the sidecar container's name
+	Sidecar string `json:"sidecar,omitempty"`
```

Comment:
> Why is this exposed by the API at all? My interpretation of #13138 is that you wanted a container-based plugin mechanism. Kubelet plugins shouldn't be exposed via the user-facing API.

+1

This is really the name of an image to used to run the rbd client in a pod. I'm not sure I would call it a sidecar container, since it runs in a dedicated pod.  I agree that we should nit expose this information via the API. I like the direction the the factoring in the volume subsystem is going, but the API changes should go away.  Instead this should be a Kubelet property.

@rootfs